File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_WinPropertyTexId.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_WinPropertyTexId
#/
class t_WinPropertyTexId:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_texId = 0	#U32
	m_partId = 0	#U32
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_texId = 0
		self.m_partId = 0

	def read(self,cVariable):
		self.m_texId = cVariable.getU32()
		self.m_partId = cVariable.getU32()
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_texId)
		cVariable.putU32(self.m_partId)
		return True

File: Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_WinPropertyTexId.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_WinPropertyTexId
*/
public class t_WinPropertyTexId : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public uint	m_texId;
	public uint	m_partId;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_WinPropertyTexId() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_texId = 0;
		m_partId = 0;
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_texId = cVariable.getU32();
			m_partId = cVariable.getU32();
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_texId);
		cVariable.put(m_partId);
		return true;
	}
};

